ddtrace/tracer/log.go

```diff
@@ -56,14 +56,14 @@ type startupInfo struct {
 // checkEndpoint tries to connect to the URL specified by endpoint.
 // If the endpoint is not reachable, checkEndpoint returns an error
 // explaining why.
-func checkEndpoint(endpoint string) error {
+func checkEndpoint(client *http.Client, endpoint string) error {
```

Comment:
I wonder if this should just be a method of the `*config`, e.g.

```
// checkEndpoint tries to connect to the configured endpoint URL.
// Returns an error if the endpoint is not reachable, nil otherwise.
func (c *config) checkEndpoint() error {
   // use c.httpClient and c.transport.endpoint()
   ...
}
```